core/src/main/java/gov/nist/secauto/metaschema/core/util/DeleteOnShutdown.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-FileCopyrightText: none
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+package gov.nist.secauto.metaschema.core.util;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Used to perform cleanup on shutdown.
+ */
+@SuppressWarnings("PMD.DoNotUseThreads")
+public final class DeleteOnShutdown {
+  private static Set<Path> paths = new LinkedHashSet<>();
+  private static final Lock LOCK = new ReentrantLock();
+
+  static {
+    Runtime.getRuntime().addShutdownHook(
+        new Thread(DeleteOnShutdown::shutdownHook));
+  }
+
+  @SuppressWarnings("PMD.NullAssignment")
+  private static void shutdownHook() {
+    LOCK.lock();
+    try {
+      Set<Path> localSet = new LinkedHashSet<>(paths);
+      paths = null;
+      localSet.forEach(path -> {
+        try {
+          Files.walkFileTree(path,
+              new SimpleFileVisitor<>() {
+                @Override
+                public FileVisitResult postVisitDirectory(
+                    Path dir, IOException exc) throws IOException {
+                  Files.delete(dir);
+                  return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFile(
+                    Path file, BasicFileAttributes attrs)
+                    throws IOException {
+                  Files.delete(file);
+                  return FileVisitResult.CONTINUE;
+                }
+              });
+        } catch (IOException ex) {
+          // this is a best effort
```

Comment:
Non-blocking: perhaps warn that DeleteOnShutdown behavior is best effort and cleanup was not complete successful, just to be on the safe side? Could help us debug later if it comes to that.

Reply:
I don't think this is possible due to how the shutdown works. Logging might be shutdown already.

Reply:
Understandable, I agree with resolving the comment and moving along.